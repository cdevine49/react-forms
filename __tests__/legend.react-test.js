import React from 'react';
import Legend from '../src/components/legend';
import renderer from 'react-test-renderer';

describe('Legend', () => {
  test("without child renders null", () => {
    const tree = renderer.create(
      <Legend
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders with child", () => {
    const tree = renderer.create(
      <Legend>Test Legend</Legend>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("accepts props", () => {
    const tree = renderer.create(
      <Legend
        id="legendId"
        className="legend-class test"
        >
        Label Fieldset
      </Legend>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});