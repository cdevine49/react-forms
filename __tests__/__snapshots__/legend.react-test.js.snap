// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend accepts props 1`] = `
<legend
  className="legend-class test"
  id="legendId"
>
  Label Fieldset
</legend>
`;

exports[`Legend renders with child 1`] = `
<legend>
  Test Legend
</legend>
`;

exports[`Legend without child renders null 1`] = `null`;
